internal-ingress-basic/README.md

```diff
@@ -0,0 +1,127 @@
+
+
+```
+$ kubectl get deploy foo
+NAME   READY   UP-TO-DATE   AVAILABLE   AGE
+foo    3/3     3            0           28s
+
+$ kubectl describe ingress foo-internal
+Name:             foo-internal
+Namespace:        default
+Address:          10.200.10.218
+Default backend:  default-http-backend:80 (10.200.20.7:8080)
+Rules:
+  Host             Path  Backends
+  ----             ----  --------
+  foo.example.com  
+                      foo:80 (<none>)
+Annotations:
+  ingress.kubernetes.io/target-proxy:                k8s2-tp-odw4qjdd-default-foo-internal-uw14fm1v
+  ingress.kubernetes.io/url-map:                     k8s2-um-odw4qjdd-default-foo-internal-uw14fm1v
+  kubectl.kubernetes.io/last-applied-configuration:  {"apiVersion":"networking.k8s.io/v1beta1","kind":"Ingress","metadata":{"annotations":{"kubernetes.io/ingress.class":"gce-internal"},"name":"foo-internal","namespace":"default"},"spec":{"rules":[{"host":"foo.example.com","http":{"paths":[{"backend":{"serviceName":"foo","servicePort":80}}]}}]}}
+
+  kubernetes.io/ingress.class:            gce-internal
+  ingress.kubernetes.io/backends:         {"k8s1-e4c0c439-default-foo-80-f3d4a56a":"Unknown","k8s1-e4c0c439-kube-system-default-http-backend-80-dee9de2e":"HEALTHY"}
+  ingress.kubernetes.io/forwarding-rule:  k8s2-fr-odw4qjdd-default-foo-internal-uw14fm1v
+Events:
+  Type     Reason     Age                  From                     Message
+  ----     ------     ----                 ----                     -------
+  Normal   ADD        23m                  loadbalancer-controller  default/foo-internal
+  Warning  Translate  23m                  loadbalancer-controller  error while evaluating the ingress spec: could not find service "default/foo"
+  Normal   CREATE     22m                  loadbalancer-controller  ip: 10.200.10.218
+  Normal   XPN        22m (x4 over 23m)    loadbalancer-controller  Firewall change required by network admin: `gcloud compute firewall-rules create k8s-fw-l7--e4c0c439b5effdb3 --network vpc-prod --description "GCE L7 firewall rule" --allow tcp:30000-32767,tcp:8080 --source-ranges 130.211.0.0/22,209.85.152.0/22,209.85.204.0/22,35.191.0.0/16 --target-tags gke-egr-airbus-test-cluster-1-f53dc7d4-node --project myproject`
+  Normal   XPN        5m16s (x3 over 15m)  loadbalancer-controller  Firewall change required by network admin: `gcloud compute firewall-rules update k8s-fw-l7--e4c0c439b5effdb3 --description "GCE L7 firewall rule" --allow tcp:30000-32767,tcp:8080 --source-ranges 130.211.0.0/22,209.85.152.0/22,209.85.204.0/22,35.191.0.0/16 --target-tags gke-egr-airbus-test-cluster-1-f53dc7d4-node --project myproject`
+```
+
+Please note in the event logs that some firewall rules should be manually configured for health checks.
+
+4. Finally, we can validate the data plane by sending traffic to our Ingress VIP.
```

Comment:
To provide everything that the user needs to complete the example, it might make sense to add in a step where an internal VM is deployed in the same zone as the cluster. We did that in the docs here: https://cloud.google.com/kubernetes-engine/docs/how-to/internal-load-balance-ingress#step_5_validate_successful_ingress_deployment

Reply:
I sumitted a new commit to take this into account (both test VM creation and deletion). The Internal VM should be in the same region as the cluster I think, it will work even if it is not in the same zone.

To be noted that Internal Ingress is only reachable from the region and will not be reachable from other regions (unlike Internal services of type Load Balancer with the global access annotation).